service_discovery_manager/Component/mmSH/server_runner.cpp

```diff
@@ -95,14 +95,11 @@ int ServerRunner::Run() {
                                                   (void*)mh_discovery_server,
                                                   OnDiscoveryServerEvent);
 
-  // TODO: MonitorServer for Android should be implemented.
-#if !defined(ANDROID)
   MonitorServer* monitor_server = new MonitorServer(UUIDS_MDS);
```

Comment:
It seems that MonitorServer now supports Android. 
Therefore, MonitorServer should be freed in any case. 
I suggest to remove #if !defined(ANDROID) in line 144.

Reply:
Thanks! I've removed line 144.